tests/test_path_resolution.py

```diff
@@ -184,3 +184,28 @@ def handler_fn(a: int = 0, b: int = 0, c: int = 0, d: int = 0) -> None:
 
     with pytest.raises(ImproperlyConfiguredException):
         create_test_client(handler_fn)
+
+
+@pytest.mark.parametrize(
+    "handler_path, request_path, expected_status_code, expected_param",
+    [
+        ("/name:str/{name:str}", "/name:str/test", HTTP_200_OK, "test"),
```

Comment:
I don't think this should be an `HTTP_200_OK`, rather `name:str` should raise an `ImproperlyConfiguredException` because this is a clear situation where the user forgot the place the value inside brackets - `{...}`.

Reply:
Not sure it is possible to know that in all cases, as "name:str" is well within the domain of valid strings that can appear in a URL. Previously, this route path got parsed as "/{name:str}" and this test case is to assure that doesn't happen any more.